src/routes/+layout.svelte

```diff
@@ -13,17 +14,23 @@
 		['/settings', 'Settings']
 	];
 
-	let theme = 'light';
+	$: theme = $settingsStore?.userTheme;
 
 	function toggleTheme() {
 		theme = theme === 'light' ? 'dark' : 'light';
 		document.documentElement.setAttribute('data-color-theme', theme);
+		settingsStore.update((store) => {
+			store!.userTheme = theme;
+			return store;
+		});
 	}
 
 	onMount(() => {
-		// Set initial theme based on user's browser preference
-		const prefersDark = window.matchMedia('(prefers-color-scheme: dark)').matches;
-		theme = prefersDark ? 'dark' : 'light';
+		if (!theme) {
+			// Set initial theme based on user's browser preference
+			const prefersDark = window.matchMedia('(prefers-color-scheme: dark)').matches;
+			theme = prefersDark ? 'dark' : 'light';
+		}
 		document.documentElement.setAttribute('data-color-theme', theme);
 	});
```

Comment:
> when user reloads the page the initial theme is light and then it turns to dark (because it gets changed onMount)

I think the correct way of doing this is not to use `onMount()` and instead use: [`import { browser } from '$app/environment';`](https://kit.svelte.dev/docs/modules#$app-environment-browser)

Here's an example ([Source](https://dev.to/willkre/persistent-theme-switch-dark-mode-with-svelte-sveltekit-tailwind-1b9g)):

```svelte
<script lang="ts">
    import { browser } from '$app/environment';

    let darkMode = true;

    function handleSwitchDarkMode() {
        darkMode = !darkMode;

        localStorage.setItem('theme', darkMode ? 'dark' : 'light');

        darkMode
            ? document.documentElement.classList.add('dark')
            : document.documentElement.classList.remove('dark');
    }

    if (browser) {
        if (
            localStorage.theme === 'dark' ||
            (!('theme' in localStorage) && window.matchMedia('(prefers-color-scheme: dark)').matches)
        ) {
            document.documentElement.classList.add('dark');
            darkMode = true;
        } else {
            document.documentElement.classList.remove('dark');
            darkMode = false;
        }
    }
</script>
```

In another project I had a similar implementation that used this technique:
https://github.com/search?q=repo%3Afmaclen%2Fcanutin-desktop%20colorThemeStore&type=code

But that alone won't solve the issue with the FOUC (_Flash of Unstyled Content_), there is one more piece to the puzzle in [`app.html`](https://github.com/fmaclen/canutin-desktop/blob/453d1869c0edbe3a707b57c7645d4bedc80ed90e/sveltekit/src/app.html#L9-L16) where we read the `localStorage` outside of Svelte.

You can try it out in the live demo: https://demo.canutin.com/settings

Reply:
I tried these options and I came to a solution where the initial theme is selected in [`app.html`](https://github.com/fmaclen/hollama/blob/20b39847bfc324c884a15f4ccd2ea36a147500e5/src/app.html#L11-L20). Then, [`+layout.svelte`](https://github.com/fmaclen/hollama/blob/20b39847bfc324c884a15f4ccd2ea36a147500e5/src/routes/%2Blayout.svelte#L18-L25) only handles the theme switch. This removes the FOUC and works correctly.